### PR TITLE
Feature/578 navbar

### PR DESCRIPTION
**Pull request by Michiel-s**

### CODE REVIEW

**Comment by Michiel-s** [comment]
@hanjoosten somehow the (ExecEngine) rules as defined in Navbar.adl (included in SystemContext.adl) are not compiled into the target context/fspec. At least I don't see them in the json output files. Do you understand why not? What is happening?

**Comment by Michiel-s** [comment]
Great. I wasn’t sure if I could change the dirtyId function, because the meat grinder also uses this function. Thx

**Comment by hanjoosten** [comment]
> @hanjoosten somehow the (ExecEngine) rules as defined in Navbar.adl (included in SystemContext.adl) are not compiled into the target context/fspec. At least I don't see them in the json output files. Do you understand why not? What is happening?

Do you compile with the `--proto` switch? Last month a change was made that the SystemContext stuff is only generated into the script if a prototype is being generated ( `--proto` switch). By doing so, documentation generated (by using the `--fspec` switch) does not show the SystemContext stuff, which was confusing for students.

**Comment by Michiel-s** [comment]
> > @hanjoosten somehow the (ExecEngine) rules as defined in Navbar.adl (included in SystemContext.adl) are not compiled into the target context/fspec. At least I don't see them in the json output files. Do you understand why not? What is happening?
> 
> Do you compile with the `--proto` switch? Last month a change was made that the SystemContext stuff is only generated into the script if a prototype is being generated ( `--proto` switch). By doing so, documentation generated (by using the `--fspec` switch) does not show the SystemContext stuff, which was confusing for students.

Yes, I use the --proto switch.

**Comment by Michiel-s** [comment]
@hanjoosten could you check commit cb7382c17c357d70ec295080108095df7af5ab28 for me. Is this ok? I removed the optionalQuote part because it conflicts with the meta population use case, but I cannot completely oversee it's effect elsewhere

**Comment by hanjoosten** [comment]
See [this comment](https://github.com/AmpersandTarski/Ampersand/commit/cb7382c17c357d70ec295080108095df7af5ab28)

**Comment by Michiel-s** [comment]
@hanjoosten can you solve the merge conflict with development? Something has changed in deamon branch in CreateFspec and this one is to hard for me to solve.

**Comment by Michiel-s** [comment]
Thx

File: ReleaseNotes.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## unreleased changes
 * [Issue #923](https://github.com/AmpersandTarski/Ampersand/issues/923) Lexer error message had disappeared. Now they show again.
 * [Issue #925](https://github.com/AmpersandTarski/Ampersand/issues/925) Add warning when a script contains `BOX <ROWSNH>`, for this is deprecated and could result in an error when a prototype is being generated.
-* [Issue #578](https://github.com/AmpersandTarski/Ampersand/issues/578) Implement meta model for navigation menu
```
Comment (hanjoosten) [inline ReleaseNotes.md]: Why was the line about Issue#578 removed? It should be put back in, I guess.

Comment (RieksJ) [reply ReleaseNotes.md]: Sorry. My mistake. I reinserted it again.

**Comment by Michiel-s** [comment]
I think this feature is ready for merge and release.

**Comment by Michiel-s** [comment]
Oh wait, I do need to check the right default prototype framework version statement.

**Pull request merged**